### ISSUE TITLE
Octatonic scale degrees

### ISSUE THREAD

**Issue by MarkGotham** [cuthbertLab/music21#755]
```
ots = scale.OctatonicScale('C')
rn = roman.RomanNumeral(8, ots)
rn.scaleDegree
```

Should be: 8
Return: 7

I don't suppose many people are loosing sleep over this, but all the same ...

**Comment by jacobtylerwalls** [comment]
Setting this to VI{0,3} and vi{0,3} takes care of it. But should we?
https://github.com/cuthbertLab/music21/blob/335ba3ed62f002bc625fea8433ab2e7fc22b4932/music21/roman.py#L1441

There's also a pretzel of pitch issues here, because exotic romans like this example from docs is okay:
```
>>> roman.RomanNumeral('I9', ots).pitches
(<music21.pitch.Pitch C#4>, <music21.pitch.Pitch E4>, <music21.pitch.Pitch G-4>,
<music21.pitch.Pitch A4>, <music21.pitch.Pitch C5>)
```
but not this one:
```
>>> roman.RomanNumeral('I', ots).pitches
(<music21.pitch.Pitch C4>, <music21.pitch.Pitch E4>, <music21.pitch.Pitch G4>)
```

I suppose `_matchAccidentalsToQuality()` shouldn't run if the useScale is not a DiatonicScale. Thoughts?

**Comment by mscuthbert** [comment]
Even though I did say "scaleCardinality" is an attribute of RomanNumerals, I'm inclined to say that RN analysis is a function of diatonic music and not fix unless there's a mainstream analysis of Octatonic music that uses RNs.  Over the years I've changed my mind into thinking that allowing people to do, say RN analysis of non-Western music, encourages people to make other people do it; or deny the differences.

**Comment by MarkGotham** [comment]
Fair enough. So then the task is to remove this in time for version 7 I guess (rather than fix in place?)

**Comment by jacobtylerwalls** [comment]
Something like this would be a soft transition -- then, assuming no one pipes up on the list on account of their repo hammer/everything-is-a-nail that does octatonic RN analysis, we can fully deprecate afterward.

```diff --git a/music21/roman.py b/music21/roman.py
index 54f606d0f..8a890862f 100644
--- a/music21/roman.py
+++ b/music21/roman.py
@@ -2542,6 +2542,9 @@ class RomanNumeral(harmony.Harmony):
         if 'DiatonicScale' in useScale.classes:  # speed up simple case
             self.scaleCardinality = 7
         else:
+            # TODO: fully deprecate, raise RomanNumeralException
+            environLocal.warn(f'RomanNumeral analysis in nondiatonic contexts
+                (got {useScale}) will be deprecated in future versions')
             self.scaleCardinality = useScale.getDegreeMaxUnique()
```

Mark, would you be keen to add a test, improve wording etc, and get this in?
